### ISSUE TITLE
Suppress warnings

### ISSUE THREAD

**Issue by brechtm** [michaeljones/breathe#568]
Sphinx supports suppressing certain types of warnings by means of the [suppress_warnings](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings) configuration variable. This doesn't to be supported by Breathe. Is there any other way to suppress warnings such as "Member ... is not documented"?

**Comment by mpusz** [comment]
Yes, I have the same issue. I am not sure if I understand which ones are being generated by Sphinx and which ones are from Breathe but even with Sphinx run with `-Q` I get a lot of:

```
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1a61ea2e38f68b3914936478613af64834
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1a303d21cd805b9d658c0aea57f1e614c2
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1aad88ac35f098e2fd546b11ce1db25287
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1aeeb7c93afa4a3464d182bc97c827769c
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1aa1c6f081e6a3010f8795163ab7d92066
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1a5ba5ec9ab4526d6abdf36fe0993ea2b2
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1a484be4c1d13037d215a813e670efc238
[build] Warning: Duplicate target detected: namespaceunits_1_1physical_1_1si_1_1literals_1af9e528ff923d83bd1ab9877f7350035a
...
```

Without `-Q` Travis CI refuses to finish the build with the message:

```
The job exceeded the maximum log length, and has been terminated.
```

More info on Travis CI issues here: https://travis-ci.com/github/mpusz/units/builds/184157816.

**Comment by utzig** [comment]
The `The job exceeded the maximum log length, and has been terminated.` might be new, but you seemed to have been having documentation issues for quite a long time. I am looking at some 4 month old build here: https://travis-ci.com/github/mpusz/units/jobs/335751297 and the exact same issues exist. Those copied messages are coming from Breathe 
due to an exception in docutils but there are also lots of messages coming from Sphinx. This is quite hard to debug for a big project, but the issue could be coming from from the Doxygen XML generation, or Sphinx/Breathe itself; if I get some free time I will try to find out.

**Comment by mpusz** [comment]
This is a Modern C++ code using a lot of C++20 features. This causes a lot of issues in all documentation generation tools. Doxygen generates only a few warnings but the quality of produced XML file leaves a lot of room for improvement. Sphinx and Breathe have a lot of issues too.

**Comment by mpusz** [comment]
> if I get some free time I will try to find out.

Please let me know in case I can help you with anything. The generated documentation can be found here: https://mpusz.github.io/units. In the https://mpusz.github.io/units/usage.html#contributing-or-just-building-all-the-tests-examples-and-documentation chapter there is a detailed instruction on how to compile all the code and documentation.

**Comment by vermeeren** [comment]
As just posted in #485, this may be useful for those with build failures due to big job logs:

> A workaround could be to write a wrapper script around the CI scripts that filters out `Warning: Duplicate target detected:` using `grep` or similar. Not a a proper fix but at least avoids build failure.

**Comment by dylanh724** [comment]
Would love to see warning suppression systems, if even basic. I can't seem to manually set the logging level, either. Any workaround ideas?

There are numerous reasons to suppress I won't digress into, but this is a pretty important feature to have. I currently have 1000 errors that I don't care too much about for those reasons beyond the scope of this post.
